### PR TITLE
Enable compatibility with Azure OpenAI

### PR DESCRIPTION
**Pull request by drdavella**

### CODE REVIEW

File: plugins/codemodder-plugin-llm/src/main/java/io/codemodder/plugins/llm/EnvironmentBasedModelMapper.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package io.codemodder.plugins.llm;
+
+import java.util.HashMap;
+
+/** Mapper that maps models to their deployment names based on environment variables. */
+final class EnvironmentBasedModelMapper implements ModelMapper {
+  private static final String DEPLOYMENT_TEMPLATE = "CODEMODDER_AZURE_OPENAI_%s_DEPLOYMENT";
+
+  private final HashMap<Model, String> map = new HashMap<>();
+
```
Comment (nahsra) [inline plugins/codemodder-plugin-llm/src/main/java/io/codemodder/plugins/llm/EnvironmentBasedModelMapper.java:10]: This is public even though type is not

File: plugins/codemodder-plugin-llm/src/main/java/io/codemodder/plugins/llm/OpenAIService.java [hunk 2]
```diff
@@ -1,69 +1,118 @@
...
+/** A custom service class to wrap the {@link OpenAIClient} */
 public class OpenAIService {
-  private final OpenAiApi api;
+  private final OpenAIClient api;
+  private static final int TIMEOUT_SECONDS = 90;
+  private final ModelMapper modelMapper;
 
-  public OpenAIService(final String token) {
-    this.api = OpenAiService.buildApi(token, Duration.ofSeconds(90));
+  private static OpenAIClientBuilder getBuilder(final KeyCredential key) {
```
Comment (nahsra) [inline plugins/codemodder-plugin-llm/src/main/java/io/codemodder/plugins/llm/OpenAIService.java:25]: The convention is just to call this method 'builder'

File: plugins/codemodder-plugin-llm/src/main/java/io/codemodder/plugins/llm/OpenAIService.java [hunk 3]
```diff
@@ -1,69 +1,118 @@
...
+        .clientOptions(clientOptions)
+        .credential(key);
   }
 
-  public ChatCompletionResult createChatCompletion(final ChatCompletionRequest request) {
-    return this.api
-        .createChatCompletion(request)
-        .retryWhen(new OpenAIRetryStrategy())
-        .blockingGet();
+  OpenAIService(ModelMapper mapper, final KeyCredential key) {
```
Comment (nahsra) [inline plugins/codemodder-plugin-llm/src/main/java/io/codemodder/plugins/llm/OpenAIService.java:34]: Make all args for this and method below final to be consistent

**Review by nahsra — CHANGES_REQUESTED** [review]
This looks great. Just some nitpicks stuff.

File: plugins/codemodder-plugin-llm/src/main/java/io/codemodder/plugins/llm/OpenAIService.java (hunk 2 above)
Comment (drdavella) [reply plugins/codemodder-plugin-llm/src/main/java/io/codemodder/plugins/llm/OpenAIService.java:25]: Even if it's not returning the builder for the class itself?

**Review by nahsra — APPROVED** [review]

**Pull request merged**